### PR TITLE
Requests result

### PR DESCRIPTION
**Pull request by NetanelBasal**
A new starter of a simpler API for managing request result:

```ts
const { state, config } = createState(withEntities<Todo>());

const store = new Store({ state, config, name: 'todos' });

const entities$ = store.pipe(
  selectAllEntities(),
  joinRequestResult(['todos'])
)

entities$.subscribe(result => {
  result.isLoading,
  result.isError,
  result.isSuccess.
  result.data // typed as Todo[]
  result.error
})

http.get('todos').pipe(
  tap((todos) => store.update(setEntities(todos))),
  trackRequestResult(['todos'])
)
```

There are some more TODOS if you want to help :)

### CODE REVIEW

**Comment by NetanelBasal** [comment]
@EricPoul see if you want to help here :)

File: packages/requests/src/lib/requests-result.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  key: unknown[],
+  options?: Options<TData>
+): MonoTypeOperatorFunction<TData> {
+  return function (source: Observable<TData>) {
+    return getRequestResult(key).pipe(
+      take(1),
+      switchMap((result) => {
+        if (result.isSuccess) {
+          return EMPTY;
+        }
```
Comment (EricPoul) [inline packages/requests/src/lib/requests-result.ts:143]: I guess we should provide an opportunity to skip the caching even if `result` is already `isSuccess`.

**Comment by EricPoul** [comment]
I can get to converting `interface Result` to the union type. Depends on how fast you want this pr done.

File: packages/requests/src/lib/requests-result.ts (hunk 1 above)
Comment (NetanelBasal) [reply packages/requests/src/lib/requests-result.ts:143]: Correct

**Comment by NetanelBasal** [comment]
No rush

**Comment by EricPoul** [comment]
Should I make a preliminary commit directly in this branch?

**Comment by NetanelBasal** [comment]
I don't think you can

**Comment by NetanelBasal** [comment]
What do you mean? data is the result of the prev selector.

**Comment by EricPoul** [comment]
I'm about this `TODO` 
```// TODO - Make stricter type using overloads. For example, if `status` is loading `data` is undefined```
Since `data` is the result of the previous selector it'll always be there no matter what the status of the request is. So that we can add `data` to the return type of the `joinRequestResult` function(`Result & { data: T }`)

**Comment by NetanelBasal** [comment]
You are right.

**Comment by EricPoul** [comment]
I've made some changes/ Please take a look https://github.com/ngneat/elf/pull/364

**Pull request merged**